components/data-handler.js

```diff
@@ -38,23 +38,27 @@ export default function dataHandler(WrappedComponent, { title, url }) {
     }
 
     search(text) {
+      const keys = [
+        'name',
+        'description',
+        'notes',
+        'needs',
+        'email',
+        'phone',
+        'address',
+        'donationNeeds',
+        'volunteerNeeds',
+      ]
       const options = {
         shouldSort: true,
         minMatchCharLength: 1,
         threshold: 0.3,
-        keys: [
-          'name',
-          'description',
-          'notes',
-          'needs',
-          'email',
-          'phone',
-          'address',
-          'donationNeeds',
-          'volunteerNeeds',
-        ],
+        keys: this.state.items[0].type ? keys.concat('type') : keys,
```

Comment:
I think this may fail if first item in list doesnt have a `type` right?

Reply:
Each sections returns a different objects in an array. The Recovery section has a `type` property, but I haven't seen it with other ones. I'm duck-typing so to speak the first one and making the assumption that all objects are consistent in the array. I can test that all objects in the array should contain the `type` property, but that's not the correct solution either for this problem. Maybe I can just include type, and `fuse` would fail gracefully if it doesn`t encounter that property?

Reply:
I was thinking that you could just check "does at least one item in the array have a `type` key" instead?

Reply:
Do you mean check a random index? Wouldn't that be the same assumption as index 0 or otherwise?